formats/json-tests/jsTest/src/kotlinx/serialization/json/DynamicSerializerTest.kt

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright 2017-2024 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization.json
+
+import kotlin.test.*
+
+class DynamicSerializerTest {
+    @Test
+    fun happyPath() {
+        val value = js("{a: 42}")
+        assertEquals("""{"a":42}""", JSON.stringify(Json.encodeToDynamic(DynamicSerializer, value)))
+        assertEquals("""{"a":42}""", JSON.stringify(Json.decodeFromDynamic(DynamicSerializer, value)))
+    }
```

Comment:
I’m happy to write a ton of tests for this. I think it needs tests for various inputs and also various JSON encoder types (dynamic, stream, tree)